parsers/ClinGenGeneDiseaseValidity/src/loadClinGenGeneDiseaseValidity.py

```diff
@@ -0,0 +1,105 @@
+        """
+        super().__init__(test_mode=test_mode, source_data_dir=source_data_dir)
+        # Issue with the file path, need to check this out
+        self.data_url = "https://search.clinicalgenome.org/kb/gene-validity/"
+        self.gene_disease_data_file = "download"
+        self.data_files = [self.gene_disease_data_file]
+
+    def get_latest_source_version(self) -> str:
+        # No version is availble at the source, using the date as versioning proxy
+        latest_version = date.today().strftime("%Y%m")
+        return latest_version
+
+    def get_data(self) -> bool:
+        # get_data is responsible for fetching the files in self.data_files and saving them to self.data_path
+        source_data_url = f"{self.data_url}{self.gene_disease_data_file}"
+        data_puller = GetData()
+        data_puller.pull_via_http(source_data_url, self.data_path)
+        return True
+
+    def parse_data(self) -> dict:
+        """
+        Parses the data file for graph nodes/edges
+
+        :return: ret_val: load_metadata
+        """
+        extractor = Extractor(file_writer=self.output_file_writer)
+        gene_disease_data_file: str = os.path.join(
+            self.data_path, self.gene_disease_data_file
+        )
+        # print(os.path.join(self.data_path, self.gene_disease_data_file)) for_debugging
+        # Need to incorporate the logic to skip the intial metadata rows
+        # either skipped record counter or using loop, rn the normaliztion version is working fine
+        with open(gene_disease_data_file, "rt") as fp:
+            extractor.csv_extract(
+                fp,
+                lambda line: f"{line[ClinGenGeneDiseaseValidityCOLS.GENE_ID.value]}",  # subject id
+                lambda line: f"{line[ClinGenGeneDiseaseValidityCOLS.DISEASE_ID.value]}",  # object id
+                lambda line: "gene_associated_with_condition",  # predicate extractor
+                lambda line: {},  # subject properties
+                lambda line: {},  # object properties
+                lambda line: {
+                    "Mode_of_Inheritance": line[
+                        ClinGenGeneDiseaseValidityCOLS.MOI.value
+                    ],
+                    "Classification": line[
+                        ClinGenGeneDiseaseValidityCOLS.CLASSIFICATION.value
+                    ],
+                    "Classification_Date": line[
+                        ClinGenGeneDiseaseValidityCOLS.CLASSIFICATION_DATE.value
+                    ],
+                    "Classification_Report": line[
+                        ClinGenGeneDiseaseValidityCOLS.ONLINE_REPORT.value
+                    ],
+                },  # edge properties
+                comment_character="#",
+                delim=",",
+                has_header_row=True,
+            )
+        return extractor.load_metadata
+
+
+# We can't directly normalize the Mode of Inheritance or MOI , as the data in the file is not suffice enough to get the MOI
```

Comment:
I thought we had worked out a mapping for MOI for these (at least with only a few exceptions).